packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/TransferERC20.tsx

```diff
@@ -0,0 +1,183 @@
+      }
+
+      let data = erc20Transfer.encodeFunctionData('transfer', [toAddress, amount]);
+      if (useTransferFrom) {
+        data = erc20Transfer.encodeFunctionData('transferFrom', [fromAddress, toAddress, amount]);
+      }
+
+      setParams([{
+        from: fromAddress,
+        to: contractAddress,
+        value: '0',
+        data,
+      }]);
+    } catch (err) {
+      setParams([{
+        from: fromAddress,
+        to: contractAddress,
+        value: '0',
+        data: '0x',
+      }]);
+    }
+  }, [fromAddress, toAddress, contractAddress, amount, erc20Transfer, useTransferFrom]);
+
+  useEffect(() => {
+    const getAddress = async () => {
+      if (zkEvmProvider) {
+        const [walletAddress] = await zkEvmProvider.request({
+          method: 'eth_requestAccounts',
+        });
+        setFromAddress(walletAddress || '');
+      }
+    };
+
+    getAddress().catch(console.log);
+  }, [zkEvmProvider, setFromAddress]);
+
+  const handleSubmit = useCallback(async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    await handleExampleSubmitted({
+      method: 'eth_sendTransaction',
+      params,
+    });
+  }, [params, handleExampleSubmitted]);
+
+  return (
+    <Accordion.Item eventKey="3">
+      <Accordion.Header>Transfer ERC20</Accordion.Header>
+      <Accordion.Body>
+        <Form noValidate onSubmit={handleSubmit} className="mb-4">
+          <Form.Group className="mb-3">
+            <Form.Label>
+              Preview
+            </Form.Label>
+            <Form.Control
+              readOnly
+              as="textarea"
+              rows={7}
+              value={JSON.stringify(params, null, '\t')}
+              style={{
+                fontSize: '0.8rem',
+              }}
+            />
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label>
+              From
+            </Form.Label>
+            <Form.Control
+              required
+              disabled={!useTransferFrom || disabled}
+              type="text"
+              value={fromAddress}
+              onChange={(e) => setFromAddress(e.target.value)}
+            />
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Check
+              onClick={() => {
+                setUseTransferFrom(!useTransferFrom);
+              }}
+              type="checkbox"
+              label="Use transferFrom method"
+            />
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label>
+              To
+            </Form.Label>
+            <Form.Control
+              required
+              disabled={disabled}
+              type="text"
+              onChange={(e) => setToAddress(e.target.value)}
+            />
+          </Form.Group>
+          <Form.Group className="mb-3">
+            <Form.Label>
+              Contract Address
+            </Form.Label>
+            <Form.Control
```

Comment:
we can preset some default contract like Spending cap did: https://github.com/immutable/ts-immutable-sdk/blob/main/packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/SpendingCapApproval.tsx#L33
did

Reply:
done 👍